py/desispec/workflow/batch.py

```diff
@@ -76,6 +84,44 @@ def default_system(jobdesc=None, no_gpu=False):
 
     return name
 
+def parse_reservation(reservation, jobdesc):
+    """
+    Parse reservation name into cpu/gpu reservation based upon jobdesc
+
+    Args:
+        reservation (str): resvname or resvname_cpu,resvname_gpu or None
+        jobdesc (str): job description string e.g. 'arc', 'flat', 'tilenight'
+
+    Returns:
+        cpu_reservation_name, gpu_reservation_name
+
+    If a single reservation name is provided, return both cpu/gpu as the same.
+    If either is 'none' (case-insensitive), return None for that reservation
+    """
+    if reservation is None:
+        return reservation
+
+    tmp = reservation.split(',')
+    if len(tmp) == 1:
+        reservation_cpu = reservation_gpu = reservation
+    elif len(tmp) == 2:
+        reservation_cpu, reservation_gpu = tmp
+    else:
+        raise ValueError(f'Unable to parse {reservation} as rescpu,resgpu')
+
+    if reservation_cpu.lower() == 'none':
+        reservation_cpu = None
+
+    if reservation_gpu.lower() == 'none':
+        reservation_gpu = None
+
+    system_name = default_system(jobdesc)
```

Comment:
Given that the cpu/gpu seperation is somewhat specific to how Perlmutter queues are partitioned, it's not obvious to me that we need to generalized this to checking the configuration for GPU's. We could just use the system name, which is either perlmutter_cpu or perlmutter_gpu.

For NERSC10 the partitions may be divided differently in a way that requires a refactor anyway.

I don't feel very strongly about this, so if you do then it is fine as-is. My proposal simplifies the need for getting and caching the config file, but otherwise is slightly less generic.

Reply:
I agree that this is a step more than we need for Perlmutter itself, but I think it gives us slightly more flexibility for the future, though you are right that NERSC-10 might throw us a monkey wrench either way.  Note that if we specify just a single reservation name it is the only one used so we at least have that fallback.  I suggest leaving this as-is for now.